File: com.issue.tracker.infra/src/main/java/com/issue/tracker/infra/logger/LoggerBuilderImpl.java
package com.issue.tracker.infra.logger;

import com.issue.tracker.api.logger.Log;
import com.issue.tracker.api.logger.LogType;
import com.issue.tracker.api.logger.LoggerBuilder;
import com.issue.tracker.api.logger.LoggerBuilderBase;
import jakarta.ejb.Stateless;

import java.io.Serializable;

@Stateless
public class LoggerBuilderImpl extends LoggerBuilderBase implements LoggerBuilder, Serializable {
    @Override
    public LoggerBuilderBase create(Class<?> invokerClass, LogType logType, String message) {
        this.invokerClass = invokerClass;
        this.logType = logType;
        this.message = message;
        return this;
    }

    @Override
    public Log build() {
        if (invokerClass == null) {
            throw new RuntimeException("Invoker class is not set");
        }
        CommonLog log = new CommonLog(invokerClass, logType, message);
        if (stackTrace != null) {
            log.setStackTrace(stackTrace);
        }
        if (reason != null) {
            log.setReason(reason);
        }
        return log;
    }
}

File: com.issue.tracker.infra/src/main/kotlin/com/issue/tracker/infra/logger/LoggerBuilderImpl.kt
package com.issue.tracker.infra.logger

import com.issue.tracker.api.logger.Log
import com.issue.tracker.api.logger.LogType
import com.issue.tracker.api.logger.LoggerBuilder
import jakarta.ejb.Local
import jakarta.ejb.Singleton
import kotlin.reflect.KClass

@Singleton
@Local
open class LoggerBuilderImpl : LoggerBuilder() {
    override fun create(
        invokerClass: KClass<*>,
        logType: LogType,
        message: String
    ): LoggerBuilder {
        this.invokerClass = invokerClass
        this.logType = logType
        this.message = message
        return this
    }

    override fun build(): Log {
        if (invokerClass == null) {
            throw Exception()
        }
        val log = CommonLog(invokerClass!!, logType, message)
        if (stackTrace != null) {
            log.stackTrace = stackTrace
        }
        if (reason != null) {
            log.reason = reason
        }
        return log
    }
}